compiler/qsc_frontend/src/typeck/check.rs

```diff
@@ -175,6 +176,16 @@ impl Checker {
         ));
     }
 
+    fn check_attr_expr(&mut self, names: &Names, expr: &Expr) {
```

Comment:
Honestly it feels weird to allow an expression here. Can we just allow a `Path` to a global if the intention is "use this alternative API"? Why would we allow lambdas? If we're allowing expressions, why not statements too? I don't think I get the use case you have in mind.